Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift

```diff
@@ -9,10 +9,52 @@ import UIKit
 open class ChatMessageListView: UITableView, Customizable, ComponentsProvider {
     private var identifiers: Set<String> = .init()
     private var isInitialized: Bool = false
-    /// Component responsible to process an array of `[ListChange<Item>]`'s and apply those changes to a view.
-    private lazy var listChangeUpdater: ListChangeUpdater = TableViewListChangeUpdater(
-        tableView: self
-    )
+
+    // MARK: - Difference Kit Snapshot Handling
+
+    // The internal properties below is to handle the DifferenceKit API. Currently it is
+    // internal because these should actually be handled in the `ChatMessageListVC` but
+    // that would make this class obsolete especially the `updateMessages(changes:)` and
+    // it would require a lot of breaking changes. So for now, the Diff logic will live here.
+
+    /// When inserting messages at the bottom, if the user is scrolled up,
+    /// we skip adding the message to the UI until the user scrolls back
+    /// to the bottom. This is to avoid message list jumps.
+    internal var skippedMessages: Set<MessageId> = [] {
+        didSet {
+            if skippedMessages.isEmpty {
+                newMessagesSnapshot = _newMessagesSnapshotWithSkipped
+            }
+        }
+    }
+
+    /// The previous messages snapshot before the next update.
+    internal var previousMessagesSnapshot: [ChatMessage] = [] {
+        didSet {
+            let isNotSkippedMessage: (ChatMessage) -> Bool = {
+                !self.skippedMessages.contains($0.id)
+            }
+            previousMessagesSnapshot = previousMessagesSnapshot.filter(isNotSkippedMessage)
+        }
+    }
+
+    /// The new messages snapshot reported by the channel or message controller.
+    private var _newMessagesSnapshotWithSkipped: [ChatMessage] = []
+    internal var newMessagesSnapshot: [ChatMessage] = [] {
```

Comment:
While testing, sth weird happened when I scrolled to the top and scrolled down fast. Only 5 messages were shown all the time (in the channel `hello everyone` that has much more) - it seems we might have an edge case in this logic. My assumption is that the skipped messages were not cleared?

Reply:
Do you mean, you couldn't load more messages? Or for a split second, you only saw 5 messages? Because the CATransition hack, sometimes doesn't work 100% and there could be animation glitches

Reply:
No, I had only 5 messages the whole time. I had to leave the channel and come back to get it back to working. Will try to have proper steps on the mob testing session.

Reply:
Ah I see, this is worrying then, definitely should not happen.  I'll investigate this one 👍

Reply:
While fixing the other problem, I noticed I was not using the previous snapshot correctly! So I think the fix I did for the missing updates, also most likely fixed this one :) But on the mob testing we should be able to confirm it 👍